cmd/metacache-walk.go

```diff
@@ -194,7 +168,9 @@ func (s *xlStorage) WalkDir(ctx context.Context, opts WalkDirOptions, wr io.Writ
 				meta.name = strings.TrimSuffix(meta.name, SlashSeparator)
 				meta.name = pathJoin(current, meta.name)
 				meta.name = decodeDirObject(meta.name)
-				out <- meta
+				if forward == "" || meta.name >= forward {
```

Comment:
It is very expensive that we have to read metadata for every file we are *skipping*.

Before the skip would be done purely based on the name. Now we have to decode and clean up before we skip.

We must be able to skip forward without having to read the file.

Reply:
In fact, this is only executed when we are inside an object (it sees xl.meta and part.*). This happens only if you are listing one directory marker

Reply:
@klauspost do you concur? - do we need this PR @vadmeste - does it optimize reading xl.meta?

Reply:
I still don't see the point of the change. Are we fixing anything?

Reply:
> I still don't see the point of the change. Are we fixing anything?

No. My point is to introduce a new listing feature which is called EndAT, which means list until a given object. But this code is hard, that's why I am sending separate PRs to review. It is also hard to do EndAt without this PR (which avoids comparing xl.meta with 'forward'

Reply:
hmm, interesting can you send the entire PR? @vadmeste hard to imagine what it does :-)